adapters/slog/slog.go

```diff
@@ -0,0 +1,256 @@
+package slog
+
+import (
+	"context"
+	"errors"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"golang.org/x/exp/slices"
+	"golang.org/x/exp/slog"
+
+	"github.com/axiomhq/axiom-go/axiom"
+	"github.com/axiomhq/axiom-go/axiom/ingest"
+)
+
+var _ slog.Handler = (*Handler)(nil)
+
+const defaultBatchSize = 1000
+
+// ErrMissingDatasetName is raised when a dataset name is not provided. Set it
+// manually using the [SetDataset] option or export "AXIOM_DATASET".
+var ErrMissingDatasetName = errors.New("missing dataset name")
+
+// An Option modifies the behaviour of the Axiom handler.
+type Option func(*Handler) error
+
+// SetClient specifies the Axiom client to use for ingesting the logs.
+func SetClient(client *axiom.Client) Option {
+	return func(h *Handler) error {
+		h.client = client
+		return nil
+	}
+}
+
+// SetClientOptions specifies the Axiom client options to pass to
+// [axiom.NewClient] which is only called if no [axiom.Client] was specified by
+// the [SetClient] option.
```

Comment:
Would it be better DX if we added `SetToken` directly without needing to know the concept of a client or an org id?

Reply:
We could but I guess that is out of scope for this PR. The options are something that is duplicated code across all adapters. Maybe generic would come in handy here.

Reply:
Imo duplication is fine in this case to make it more usable